TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/environment.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import Optional
+from console_link.models.cluster import Cluster
+from console_link.models.metrics_source import MetricsSource
+from console_link.logic.metrics import get_metrics_source
+from console_link.logic.backfill import get_backfill
+from console_link.models.backfill_base import Backfill
+import yaml
+from cerberus import Validator
+
+
+logger = logging.getLogger(__name__)
+
+SCHEMA = {
+    "source_cluster": {"type": "dict", "required": False},
+    "target_cluster": {"type": "dict", "required": True},
+    "replayer": {"type": "dict", "required": False},
+    "backfill": {"type": "dict", "required": False},
+    "metrics_source": {"type": "dict", "required": False},
+}
+
+
+class Environment:
+    source_cluster: Optional[Cluster] = None
+    target_cluster: Optional[Cluster] = None
+    backfill: Optional[Backfill] = None
+    metrics_source: Optional[MetricsSource] = None
```

Comment:
If all of these are `None` is that an error, should this class have logic to check this?

Reply:
There's a requirement in the schema that target_cluster is populated, so from that direction we can be confident that `target_cluster` ends up populated and that might be worth checking at the end of the constructor.

But otherwise, I think it's possible that a user ends up here with nothing else provisioned. That's certainly an unusual case, so maybe we should throw them a warning if they don't have `backfill` OR `replay` OR `metadata_migration` defined, but in general a lot of `None`s is a very possible state.